python/neuroglancer/chunks.py

```diff
@@ -36,14 +63,20 @@ def decode_npz(string):
     fileobj = io.BytesIO(zlib.decompress(string))
     return np.load(fileobj)
 
-def encode_jpeg(subvol):
-    shape = subvol.shape
-    reshaped = subvol.reshape(shape[0] * shape[1], shape[2])
-    img = Image.fromarray(reshaped)
-    f = io.BytesIO()
-    img.save(f, "JPEG")
-    return f.getvalue()
+def encode_raw(subvol):
+    return subvol.tobytes('C')
 
+def decode_raw(bytestring, shape=(64,64,64), dtype=np.uint32):
+    # print shape, len(bytestring)
 
-def encode_raw(subvol):
-    return subvol.tostring('C')
\ No newline at end of file
+    data = np.frombuffer(bytestring, dtype=dtype)
+
+    if len(data) == (64 ** 3):
```

Comment:
Here's what can be figured out. The JPEG has a width and height. The height parameter mixes two elements of the shape, the height of an individual slice and the number of slices (y and z). The x parameter (width) can be known. We can use that to refine this heuristic, but there will still be problems related to the y and z parameters. The real fix is to ensure our pipeline only uploads valid files, then we can remove these heuristics.